docker/init-server.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/sh
+# exit when any command fails
+set -e
+
+# Create required directory structure (if it does not already exist)
+if [[ ! -d "$INVENTREE_STATIC_ROOT" ]]; then
+    echo "Creating directory $INVENTREE_STATIC_ROOT"
+    mkdir -p $INVENTREE_STATIC_ROOT
+fi
+
+if [[ ! -d "$INVENTREE_MEDIA_ROOT" ]]; then
+    echo "Creating directory $INVENTREE_MEDIA_ROOT"
+    mkdir -p $INVENTREE_MEDIA_ROOT
+fi
+
+# Check if "config.yaml" has been copied into the correct location
+if test -f "$INVENTREE_CONFIG_FILE"; then
+    echo "$INVENTREE_CONFIG_FILE exists - skipping"
+else
+    echo "Copying config file to $INVENTREE_CONFIG_FILE"
+    cp $INVENTREE_HOME/InvenTree/config_template.yaml $INVENTREE_CONFIG_FILE
+fi
+
+# Setup a dev environment on the mounted filesys
+if [[ -n "$INVENTREE_DEV_VENV" ]]; then
+    echo "Setting up a virtual env in ${INVENTREE_HOME}"
+    # Setup a virtual environment (within the "dev" directory)
+    python3 -m venv ./dev/env
+
+    # Activate the virtual environment
+    source ./dev/env/bin/activate
+
+    echo "Installing required packages..."
+    pip install --no-cache-dir -U -r ${INVENTREE_HOME}/requirements.txt
+fi
+
+if [[ "${INVENTREE_MIGRATE:-True}" = "True"  || "${INVENTREE_CREATE_SUPERUSER:-True}" = "True" ]]; then
```

Comment:
What is the advantage of running particular workflows based on the environment variables? This is different to the current docs guide - https://inventree.readthedocs.io/en/latest/start/docker_prod/ - should we support *both* approaches or will the install guide need to be re-written?

Reply:
Both work from my local tests.

The migrate and createsuperuser stage does make the startup slightly more streamlined. For scaling out to additional django servers there is a need to disable/not-run that migrate stage or when workers start/restart they would attempt to run it.

Reply:
If these vars are not set then the operation of the dockerfile / docker-compose would not otherwise be affected, right?

Reply:
In the single node configuration mentioned in the docs (where only one inventree-server runs), a user shouldnt notice a change. 
When they're not set, the `:-True` provides the default value which should match what was currently happening in the start_* scripts, or mentioned in the docs. 

- The pre-PR configuration runs the migrate stage each time unless the var INVENTREE_MIGRATE is set to something besides True
- The createsuperuser command will try and continue if the user is present, or create one if the DJANGO_SUPERUSER_* set of vars is set.  The dev_config.env now contains the defaults mentioned in the docs for the superuser. 

I tried to retain the default behavior as much as possible

Reply:
The switches mean adding extra servers to the current set of inventree-server set would look like:

    services:
        inventree-server-x:
            ...
            environment:
                INVENTREE_MIGRATE: "False"
                INVENTREE_CREATE_SUPERUSER: "False"
                ....
            command: [ "wait-for-it.sh", "--timeout=0",  inventree-server:8000", 
                                   "/init-server.sh", 
                                   "gunicorn" ... ]

This config would require setting up a loadbalancer to either: 
- route directly to the inventree-server containers instead of nginx
- route directly to inventree-server containers that integrate/contain nginx directly